neuro_san/service/http/server/http_server_app.py

```diff
@@ -137,10 +147,23 @@ def get_stats(self) -> str:
         return str(stats_dict)
 
     def log_request(self, handler):
-        request = handler.request
-        metadata: Dict[str, Any] = handler.get_metadata()
-        status = handler.get_status()
-        duration = 1000 * request.request_time()  # in milliseconds
-        # handler.logger is our custom HttpLogger
-        handler.logger.info(metadata, "%d %s %s (%s) %.2fms",
-                            status, request.method, request.uri, request.remote_ip, duration)
+        if isinstance(handler, BaseRequestHandler):
+            request = handler.request
+            metadata: Dict[str, Any] = handler.get_metadata()
+            status = handler.get_status()
+            duration = 1000 * request.request_time()  # in milliseconds
+            # handler.logger is our custom HttpLogger
+            handler.logger.info(metadata, "%d %s %s (%s) %.2fms",
+                                status, request.method, request.uri, request.remote_ip, duration)
+        elif isinstance(handler, ErrorHandler):
+            request = handler.request
+            metadata: Dict[str, Any] =\
+                BaseRequestHandler.get_request_metadata(request, self.forwarded_request_metadata)
+            status = handler.get_status()
+            duration = 1000 * request.request_time()  # in milliseconds
+            # handler.logger is our custom HttpLogger
+            self.logger.error(metadata, "%d %s %s (%s) %.2fms",
+                              status, request.method, request.uri, request.remote_ip, duration)
```

Comment:
Special twist for "standard" Tornado error handler - use our own logger with metadata extracted from incoming request. This allows to identify source of problem requests better, at least map them to client user_id.